### PR TITLE
add #dig to Schema::InputObject and Query::Arguments

### PR DESCRIPTION
**Pull request by modosc**
see also #1861

### CODE REVIEW

File: lib/graphql/dig.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+module GraphQL
+  module Dig
+    # implemented using the old activesupport #dig instead of the ruby built-in
+    # so we can use some of the magic in Schema::InputObject and Query::Arguments
+    # to handle stringified/symbolized keys.
+    #
+    # @param args [Array<[String, Symbol>] Retrieves the value object corresponding to the each key objects repeatedly
+    # @return [Object]
+    def dig(*args)
```
Comment (rmosolgo) [inline lib/graphql/dig.rb]: It looks like the first item from this list is used for `self[]`, then the rest are passed on to the resulting object, is that right? Another way to specify that would be: 

```ruby 
def dig(own_key, *rest_keys) 
```

Then, you don't have to do any confusing array modifications. (For example, `self[args.shift]` is a pretty complicated line, performing a read, an array modification, and a key lookup!)

File: lib/graphql/dig.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      val = self[args.shift]
+      if val.nil? || (args.respond_to?(:empty?) ? !!args.empty? : !args)
```
Comment (rmosolgo) [inline lib/graphql/dig.rb]: Won't `args` always be an array here? Why should we check `args.respond_to?(:empty?)`?

File: spec/graphql/query/arguments_spec.rb [hunk 3]
```diff
@@ -121,6 +121,33 @@
...
+    it "returns nil for missing keys" do
+      assert_nil arguments.dig("z")
+      assert_nil arguments.dig(7)
+    end
+
+    it "handles underscored keys" do
+      # TODO shouldn't either of these work?
+      # assert_equal 3, arguments.dig('input_object', 'd')
+      # assert_equal 3, arguments.dig(:input_object, :d)
+    end
```
Comment (rmosolgo) [inline spec/graphql/query/arguments_spec.rb]: Did you expect these to work, but they dont? Or, what do you mean by this comment?

File: lib/graphql/dig.rb (hunk 2 above)
Comment (modosc) [reply lib/graphql/dig.rb]: you're right - i just copied the code from:

https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/blank.rb#L18

i'll update it.

File: lib/graphql/dig.rb (hunk 1 above)
Comment (modosc) [reply lib/graphql/dig.rb]: good idea, i'll update this.

File: spec/graphql/query/arguments_spec.rb (hunk 3 above)
Comment (modosc) [reply spec/graphql/query/arguments_spec.rb]: hah, no, there's no reason that should work, i was just copy-pasting. will remove.

**Comment by modosc** [comment]
ok, changes made

**Comment by modosc** [comment]
ping - curious if any other changes are needed here?

**Comment by rmosolgo** [comment]
Thanks for the bump, sorry I let it slide, looks great!

**Comment by modosc** [comment]
@rmosolgo any idea when this will get released?

**Pull request merged**